### PR TITLE
SETI-520: Tune backoff parameters

### PR DESCRIPTION
**Pull request by mezis**
Changes the backoff from quadratic to exponential.

With the defaults, recovery time[1] is similar (120s v. 112s with the
original formula), but the max retry interval is much higher (60s v
3.3s) which should relieve the bus in case of bad subscribers.

In the example case described in the ticket, we should see a change from 2,400 throttled batches/second to ~130/second (a 36x improvement).

[1]: recovery time is the sum of backoff intervals during recovery - with the new defaults, 60s + 30s + 15s + ... = 120s. After recovery, throughput to the subscriber is back to being uncapped.

===

Jira story [#SETI-520](https://deliveroo.atlassian.net/browse/SETI-520) in project *Software Engineering Tools and Infrastructure*:

> ## Why
> 
> During a recent subscriber outage, the load on Routemaster caused by backoff/retries was higher than we'd like.
> 
> Currently, the highest backoff is 3,333ms; which means that during an outage, any queued back will be retried (and generally throttled) every 3,333ms.
> 
> The failing subscriber had accumulated ~8k batches, which means 2,400 throttlings per second (which have a Redis cost) were happening.
> 
> 
> ## What
> 
> Tune the backoff settings so the minimum retry interval is 500ms (instead of 0.33ms) and the maximum is ~1 minute.
> 
>  !image-2017-08-07-15-18-33-786.png|thumbnail!

### CODE REVIEW

File: routemaster/services/throttle.rb [hunk 1]
```diff
@@ -35,7 +37,7 @@ def check!(current_time)
       #
```
Comment (gregbeech) [inline routemaster/services/throttle.rb]: It looks like the comment above this method is now outdated.

File: spec/services/throttle_spec.rb [hunk 2]
```diff
@@ -125,55 +125,55 @@ def perform
...
-          expect(perform).to eql 0
+          expect(perform).to eq 0
         end
       end
 
       context "with 99 HP" do
         let(:hp) { 99 }
 
-        it "returns the right value" do
+        it "returns the rigth value" do
```
Comment (gregbeech) [inline spec/services/throttle_spec.rb]: You seem to have "corrected" this to the wrong spelling?

**Review by gregbeech — APPROVED** [review]
I've approved as the code looks good, but a couple of bits of text could do with a change.

File: spec/services/throttle_spec.rb (hunk 2 above)
Comment (mezis) [reply spec/services/throttle_spec.rb]: Probably a fumbling in Vim. Thanks!

**Comment by mezis** [comment]
Testing on staging: this significantly reduces the volume of delivery attempts for "dead" subscribers:

image

And it improves the throttling rate, although not as much as we'd have liked:

image

**Review by tompave-roo — APPROVED** [review]

**Pull request merged**